packages/xstate-solid/src/createImmutable.ts

```diff
@@ -97,7 +93,7 @@ const updateStore = <Path extends unknown[]>(
  * Based on Ryan Carniato's createImmutable prototype
  * Clones the initial value and diffs updates
  */
-export function createImmutable<T extends object | []>(
+export function createImmutable<T extends object>(
```

Comment:
The `| []` bit was important/useful~ here. But since this is not a public API, it's kinda irrelevant to us. You can observe the difference [here](https://www.typescriptlang.org/play?#code/CYUwxgNghgTiAEAzArgOzAFwJYHtXwxAGcMBGAHgBV4QAPQ1YI+HAIwCtwMA+ACigBc8SgEohlAFASweEvFgx4AXgLEyvANqkANPADkiHDj0BdERID0F+DfgA9APxTQkWAhTpseVSQBMVGnoQRmY2Tkx4AB94DRM+QWExYSkZVDkMZAAHCAQVQj9NHX1DYzNLa1tHIA)

Reply:
Yeah, I initially wrote all the types for a generic version of `createImmutable` but they end up being too broad for what we're using it for.